firedrake/slate/slate.py

```diff
@@ -242,6 +242,7 @@ class Tensor(TensorBase):
     """
 
     operands = ()
+    prec = None
```

Comment:
If you want to debug Slate and, for what ever reason, want to `print Tensor(foo)`, then this must be specified. Otherwise an `AttributeError` occurs. It's primarily used in the compiler to parenthesize, but it's also used for pretty printing.